### PR TITLE
meson support

### LINKED ISSUE #101: Fails to build in parallel
**Describe the bug**
When I try to build timeshift with `-j4` I get many errors like 
`cc1: fatal error: /var/tmp/portage/app-backup/timeshift-22.06.5/work/timeshift-22.06.5/src/Utility/AppLock.vala.c: No such file or directory
compilation terminated.`

I watched the directory `src/Utility/` during compilation and those files are indeed created but later deleted again. 
It works with `-j1`.

**System:**
 - Gentoo Linux
 - Timeshift 22.11.1 but also have the same bug with 22.06.5 and 21.09.1

### PR DESCRIPTION
**Pull request by yegorich**
My first take on meson build system

TODO:

- install resource files like `timeshift.json` etc.

Compile time with `make`: ca. 21 seconds
Compile time with `meson`: ca. 2 seconds

Fixes #101

### CODE REVIEW

**Comment by yegorich** [comment]
Build instructions:

```
meson setup build
meson compile -C build
```

Test installation:

```
DESTDIR=/tmp/ts-install meson install -C build
```

**Comment by yegorich** [comment]
Almost done. The only bigger missing bits so far are:

- [localization](https://mesonbuild.com/Localisation.html)
- man page

@CyberTailor want to take a look?

**Comment by yegorich** [comment]
@mtwebster do the langauge file have to be named `timeshift-<lang>.po` and not just `<lang>.po`? AFAIK, the files are named so in [Launchpad](https://bazaar.launchpad.net/~clementlefebvre/linuxmint/translations/files/head:/timeshift).

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    dependency('vte-2.91'),
+    dependency('gee-0.8'),
+    dependency('xapp'),
+    meson.get_compiler('vala').find_library('posix'),
+    meson.get_compiler('c').find_library('m'),
+]
+
+am_cflags = [
+    '-fstack-protector',
+    '--std=c99',
```
Comment (eli-schwartz) [inline meson.build]: Set this in the project's `default_options: ['c_std=c99']`.

File: src/meson.build [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+)
+
+executable('timeshift-gtk',
+  sources + sources_core + sources_utility + sources_gtk,
+  dependencies: dependencies,
+  install: true
+)
+
+install_data(sources : 'timeshift-launcher', install_dir : 'bin')
+install_data(sources : 'timeshift-uninstall', install_dir : 'bin')
```
Comment (eli-schwartz) [inline src/meson.build]: bin -> get_option('bindir')

File: src/meson.build [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+install_data(sources : 'timeshift-gtk.desktop', install_dir : 'share/applications')
```
Comment (eli-schwartz) [inline src/meson.build]: share/applications -> join_paths(get_option('datadir'), 'applications')

File: meson.build (hunk 1 above)
Comment (yegorich) [reply meson.build]: Fixed

File: src/meson.build (hunk 2 above)
Comment (yegorich) [reply src/meson.build]: Fixed

File: src/meson.build (hunk 3 above)
Comment (yegorich) [reply src/meson.build]: Fixed

**Comment by yegorich** [comment]
Both localization and man page creation are working now.

Program's version number is taken from the main `meson.build` file via a configuration file `config.vala`.

**Comment by yegorich** [comment]
I have also added an `.editorconfig` file. For now, it only handles meson files.

**Comment by yegorich** [comment]
xapp is now an optional dependency.

**Comment by CyberTailor** [comment]
> @CyberTailor want to take a look?

There's an automagic dependency on `xapp`. It should be controlled via `meson_options.txt`. Overall, good job :-)

**Comment by yegorich** [comment]
Fixed. Thanks for your review.

**Comment by mtwebster** [comment]
One file looks out of place:
image

default.json should be in /etc/timeshift, not /etc.
https://github.com/linuxmint/timeshift/blob/master/src/Core/Main.vala#L288

Looks pretty good otherwise.

**Comment by yegorich** [comment]
Good catch. I fixed it.

**Comment by mtwebster** [comment]
Ah I guess we should have just dropped the manpage PR :(  This needs  a rebase

**Comment by yegorich** [comment]
No problem. Fixed.

**Comment by mtwebster** [comment]
Are you good with this? I'm ready to merge it.

**Comment by yegorich** [comment]
Yes, I'm good with this. Let's merge it.

**Pull request merged**